proxy/envoy/config.go

```diff
@@ -136,32 +137,59 @@ func buildConfig(listeners Listeners, clusters Clusters, lds bool, mesh *proxyco
 	return out
 }
 
-// buildListeners produces a list of listeners and referenced clusters.
-func buildListeners(env proxy.Environment, sidecar proxy.Sidecar) (Listeners, Clusters) {
+// buildListeners produces a list of listeners and referenced clusters for all proxies
+func buildListeners(env proxy.Environment, role proxy.Role) Listeners {
+	switch role.Type {
+	case proxy.Sidecar:
+		listeners, _ := buildSidecar(env, role)
+		return listeners
+	case proxy.Ingress:
+		return buildIngressListeners(env.Mesh, role)
+	case proxy.Egress:
+		return buildEgressListeners(env.Mesh, role)
+	}
+	return nil
+}
+
+func buildClusters(env proxy.Environment, role proxy.Role) (clusters Clusters) {
+	switch role.Type {
+	case proxy.Sidecar:
+		_, clusters = buildSidecar(env, role)
```

Comment:
Okay I see the difference. But this is bit unclean. Not to mention computing listener and cluster twice.